portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -143,13 +140,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
  
     // Check to see if user follows any tags
     if (LoadFollowedTags.hasFollowedTags(email)) {
-      List<String> newTags = LoadFollowedTags.getFollowedTags(email);
-      for (String tag : newTags) {
-        if (!tagsToSearch.contains(tag)) {
-          tagsToSearch.add(tag);
-        }
-      } 
+      return tagsToSearch;
```

Comment:
i think you want to return if user doesn't have followedTags. you missed the '!' in the condition

Reply:
In that case, you may want to return a bool to indicate whether you actually update the |tagsToSearch| argument or not. Remember to add a comment to explain what the return type is.